org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTDelegateCommandHandler.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal;
+
+import java.util.List;
+
+import org.codehaus.plexus.util.StringUtils;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.ls.core.internal.corext.codemanipulation.OrganizeImportsOperation;
+import org.eclipse.jdt.ls.core.internal.corrections.InnovationContext;
+import org.eclipse.jdt.ls.core.internal.corrections.proposals.CUCorrectionProposal;
+import org.eclipse.jdt.ls.core.internal.corrections.proposals.IProposalRelevance;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import org.eclipse.ltk.core.refactoring.TextChange;
+import org.eclipse.text.edits.TextEdit;
+
+public class JDTDelegateCommandHandler implements IDelegateCommandHandler {
+
+	/* (non-Javadoc)
+	 * @see org.eclipse.jdt.ls.core.internal.IDelegateCommandHandler#executeCommand(java.lang.String, java.util.List, org.eclipse.core.runtime.IProgressMonitor)
+	 */
+	@Override
+	public Object executeCommand(String commandId, List<Object> arguments, IProgressMonitor monitor) throws Exception {
+		if (!StringUtils.isBlank(commandId)) {
+			switch (commandId) {
+				case "java.edit.organizeImports":
+					return organizeImports(arguments);
+				default:
+					break;
+			}
+		}
+		throw new UnsupportedOperationException(String.format("Java language server doesn't support the command '%s'.", commandId));
+	}
+
+	private Object organizeImports(List<Object> arguments) {
+		try {
+			String fileUri = (String) arguments.get(0);

```

Comment:
we need to be able to pass a uri that is not a compilation unit, but that rather matches a package, a source folder or a project